### PR TITLE
177349228 user supplied sbatch

### PR DESCRIPTION
**Pull request by zsinx6**
This works together with the https://github.com/omnivector-solutions/jobbergate-cli/pull/59 to enable `--sbatch-params` and inject then in the script being created

### CODE REVIEW

**Comment by jamesbeedy** [comment]
Manually tested on local with https://github.com/omnivector-solutions/jobbergate-cli/pull/59 👍

File: apps/job_scripts/views.py [hunk 1]
```diff
@@ -20,6 +20,20 @@
 # from rest_framework.parsers import FileUploadParser  # FIXME - why was this here?
 
 
+def inject_sbatch_params(job_script_data_as_string, sbatch_params):
```
Comment (corydodt) [inline apps/job_scripts/views.py:23]: please add a docstring to all modules, classes, and functions you touch (or create).

(if you see them, it's also good to add docstrings wherever you see them missing, even if you didn't touch them)

File: apps/test_jobbergate_apps/test_job_scripts_views.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import pytest
```
Comment (corydodt) [inline apps/test_jobbergate_apps/test_job_scripts_views.py:1]: module docstring

Comment (corydodt) [reply apps/test_jobbergate_apps/test_job_scripts_views.py:1]: (and docstrings for test functions)

File: apps/test_jobbergate_apps/test_job_scripts_views.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        '$SLURM_NODE_ALIASES\\necho $SLURM_CLUSTER_NAME\\necho $SLURM_JOB_CPUS_PER_NODE\\necho ' \
+        '$SLURM_JOB_PARTITION\\necho $SLURM_JOB_NUM_NODES\\necho $SLURM_JOBID\\necho ' \
+        '$SLURM_NODELIST\\necho $SLURM_NNODES\\necho $SLURM_SUBMIT_HOST\\necho $SLURM_JOB_ID\\necho ' \
+        '$SLURM_CONF\\necho $SLURM_JOB_NAME\\necho$SLURM_JOB_NODELIST"}'
+    return content
+
+
+@pytest.fixture
+def new_job_script_data_as_string():
+    content = '{"application.sh": "#!/bin/bash\\n\\n#SBATCH --comment=some_comment\\n#SBATCH --nice=-1\\n' \
```
Comment (corydodt) [inline apps/test_jobbergate_apps/test_job_scripts_views.py:19]: this is acceptable, also good use of `pytest.fixture()`.

but i would prefer if you wrote out it with line breaks and indented. the output here is a json.dumps() string it looks like, so just write something like

```
content = json.dumps({"application.sh": 
"""
#!/bin/bash

#SBATCH ....
# blah blah
""")})
```

That way it's readable.

Even better! use textwrap.dedent


    from textwrap import dedent
    
    content = json.dumps({"application.sh": dedent("""\
            #!/bin/bash
            #SBATCH ....
            # blah blah
            """)})

Comment (zsinx6) [reply apps/test_jobbergate_apps/test_job_scripts_views.py:19]: Awesome! I had never used the dedent

**Pull request merged**